### ISSUE TITLE
Require import Incompatibility with mocha 6.0.0

### ISSUE THREAD

**Issue by ezhaka** [travisjeffery/mocha-teamcity-reporter#50]
When mocha 6 is used test run fails with error:

```
node_modules/mocha-teamcity-reporter/lib/teamcity.js:88
        Base.call(this, runner);
      ^
TypeError: Cannot read property 'call' of undefined
```

`require('mocha').reporters.Base` seems to be missing in mocha 6.

**Comment by jamie-sherriff** [comment]
Hi @ezhaka i do not have this problem can you please provide an example test that shows it failing?

**Comment by tgrossen** [comment]
@ezhaka Did you ever find a solution to this? Currently facing the same issue with mocha@6.0.2 and mocha-teamcity-reporter@2.5.2.

**Comment by jamie-sherriff** [comment]
@tgrossen Can you provide an example please? i have not been able to find any tests that show this issue. 

In what context are you calling mocha within? (browser/node/webpack) etc?

**Comment by mpareja** [comment]
We are encountering the same issue upon upgrading to mocha 6. Mocha 6 appears to have reversed the order that REPORTER scripts and REQUIRE scripts are loaded. Mocha 5 loaded reporters ahead of require scripts.

We use require scripts to setup JSDOM, therefore initializing `window`. With the change in loading order, `mocha-teamcity-reporter` believes it is running in a browser. Perhaps there is a different way to detect when running under node?

**Comment by mpareja** [comment]
For the moment, we're working around this by adding a require script with the following:
```javascript
global.window.name = 'nodejs';
```

**Comment by jamie-sherriff** [comment]
Thanks for the info @mpareja, the logic to detect what the reporter us running under  is to support mocha-phantomjs which i think ill drop support for in the next version and just default to node so that would solve this issue.

@tgrossen and @ezhaka  are you guys doing something similar?

**Comment by jamie-sherriff** [comment]
@mpareja  I have published a pre release with the window if statement logic removed.
```npm install mocha-teamcity-reporter@next``` if you would like to start using it early.

**Comment by mpareja** [comment]
Thanks @jamie-sherriff. I've confirmed that `next` solves the problem for me. I was successfully able to remove the window name hack. I'll be on the look-out for the next major release.

**Comment by jamie-sherriff** [comment]
Fixed in 3.0.0 release
